src/main.rs

```diff
@@ -1,8 +1,6 @@
-use std::io::Result;
+use blaze::shell::command_handler::ShellCommandHandler;
 
-use blaze::shell;
-
-fn main() -> Result<()> {
-    shell::handling::handle_command_arguments()?;
-    Ok(())
+#[tokio::main]
+async fn main() {
```

Comment:
I think we need move main to separate crate

As far as I know, using a combined lib and executable crate has dependency issues

Reply:
I haven't heard about this approach and the problems without it. What advantages does it have?

Reply:
> I haven't heard about this approach and the problems without it. What advantages does it have?

Users of this crate (as lib) also download dependencies for shell app

Reply:
Okay, it does make sense